aten/src/ATen/WrapDimUtils.h

```diff
@@ -27,6 +28,24 @@ static inline int64_t maybe_wrap_dim(int64_t dim, int64_t dim_post_expr, bool wr
   return dim;
 }
 
+#ifndef __CUDA_ARCH__
+// on Windows, std::bitset cannot seem to cope with the
+// non-explicit half conversion in THCUNN/THCHalfAutoNumerics.cuh
+// so this is host-code only
+
+static inline std::bitset<64> dim_list_to_vector(IntList dims, int64_t ndims, bool wrap_scalar=true) {
```

Comment:
So bitset is nice but we shouldn’t assume that all our accesses will be in range. All functions that use this need to have asserts to make sure that it’s safe

Reply:
Also, wouldn’t `__host__` annotation fix the issue?

Reply:
Thank you @apaszke , @peterjc123  for your comments.
As far as I can tell, a host annotation would not fix this, as the half conversion seems to also be included for host compiling .cu files.
Also, I don't know whether the problem is actually the 64. My impression is that it is some internals within MSVC's bitset that is odd.
I'll make the 64 a constexpr as suggested.

How about sticking the `dim_list_to_vector` in a separate .h `WrapDimUtilsCpp.h` or so?

Best regards

Thomas